cranelift/filetests/src/test_zkasm.rs

```diff
@@ -53,6 +54,22 @@ mod tests {
                 _ => unimplemented!("Global type is not supported"),
             }
         }
+
+        // Generate const data segments definitions.
+        for (offset, data) in data_segments {
+            program.push(format!("  {offset} => E"));
+            // Each slot stores 8 consecutive u8 numbers, with earlier addresses stored in lower
+            // bits.
+            for (i, chunk) in data.chunks(8).enumerate() {
+                let mut chunk_data = 0u64;
+                for c in chunk.iter().rev() {
+                    chunk_data <<= 8;
+                    chunk_data += *c as u64;
```

Comment:
Not that it matters much, but if you're shifting, then consider using logical or, rather than add. Furthermore, if you do a

```
chunk_data |= c as u64 << shift
```

rather than the current implementation, then you may end up not having to deal with data dependencies in each loop iteration, which can make the execution faster.

But in either case both points are minor nits, feel free to ignore.

Reply:
Good point, I've switched to `or`, but left the rest the same, we don't need performance here, and introducing a `shift` makes the code a bit longer and trickier to read (at least the version that I came up with...)